api/turing/service/ensembling_job_service.go

```diff
@@ -0,0 +1,148 @@
+package service
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/gojek/turing/api/turing/models"
+	"github.com/jinzhu/gorm"
+)
+
+const (
+	sparkHomeFolder string = "/home/spark"
+)
+
+// EnsemblingJobFindByIDOptions contains the options allowed when finding ensembling jobs.
+type EnsemblingJobFindByIDOptions struct {
+	ProjectID *models.ID
+}
+
+// EnsemblingJobListOptions holds query parameters for EnsemblersService.List method.
+type EnsemblingJobListOptions struct {
+	PaginationOptions
+	ProjectID *models.ID
+	Status    models.Status
```

Comment:
Should this be `*models.Status` as well? You can also add validation tags, such as `validate:"oneof=pending running ..."`

Reply:
Not possible with the linter throwing errors saying that the line is too long.
```
turing/service/ensembling_job_service.go:25: line is 134 characters (lll)
        Status    *models.Status `validate:"oneof=pending running terminating terminated completed failed failed_submission failed_building"`
```

Will omit this suggestion.

Reply:
You can exclude it from linter: https://golangci-lint.run/usage/false-positives/#nolint

Reply:
Alright removed the linting error and put that line back in.